src/components/GlobeHealpix.vue

```diff
@@ -399,39 +399,43 @@ async function processDataVar(
       addOffset = -low * scaleFactor;
     }
 
-    await Promise.all([...Array(HEALPIX_NUMCHUNKS).keys()].map(async (ipix) => {
-      const texData = await getHealpixData(
-        datavar,
-        currentTimeIndexSliderValue,
-        ipix,
-        HEALPIX_NUMCHUNKS
-      );
-
-      // Update global data range
-      dataMin = Math.min(dataMin, texData.min);
-      dataMax = Math.max(dataMax, texData.max);
-
-      const geometry = makeHealpixGeometry(1, ipix, gridStep);
-      const material = makeTextureMaterial(
-        texData.texture,
-        props.colormap!,
-        addOffset,
-        scaleFactor
-      );
-
-      const mesh = new THREE.Mesh(geometry, material);
-      getScene()!.add(mesh);
-
-      if (mainMeshes[ipix]) {
-        mainMeshes[ipix] = mesh;
-      } else {
-        mainMeshes.push(mesh);
-      }
-
-      // Optional: trigger a render here if using manual render loop
-      redraw();
-      // If your render loop is auto-updating (via requestAnimationFrame), you may skip redraw()
-    }));
+    await Promise.all(
+      [...Array(HEALPIX_NUMCHUNKS).keys()].map(async (ipix) => {
+        const texData = await getHealpixData(
+          datavar,
+          currentTimeIndexSliderValue,
+          ipix,
+          HEALPIX_NUMCHUNKS
+        );
+
+        // Update global data range
+        dataMin = Math.min(dataMin, texData.min);
+        dataMax = Math.max(dataMax, texData.max);
+
+        const geometry = makeHealpixGeometry(1, ipix, gridStep);
+        const material = makeTextureMaterial(
+          texData.texture,
+          props.colormap!,
+          addOffset,
+          scaleFactor
+        );
+
+        if (mainMeshes[ipix]) {
+          mainMeshes[ipix]!.geometry.dispose();
+          mainMeshes[ipix]!.geometry = geometry;
```

Comment:
do we have to rebuild the geometry all the time, or would it be better to keep the old geometry in case the size didn't change?

-- if it's too much right now, we can also move that to an issue for later implementation.

Reply:
you are totally right. I actually already had the correct logic for geometry in place and just deleted this code